examples/vmi-with-generic-hook.yaml

```diff
@@ -0,0 +1,35 @@
+---
+apiVersion: kubevirt.io/v1
+kind: VirtualMachineInstance
+metadata:
+  annotations:
+    hooks.kubevirt.io/hookSidecars: '[{"args": ["--version", "v1alpha2"], "image":"registry:5000/kubevirt/sidecar-shim:devel",
+      "configMap": {"name": "my-config-map","key": "my_script.sh", "hook": "ondefineDomain"}}]'
```

Comment:
Aren't we missing the configmap for the example? This won't work without it, right? If so, then I would add an example of configmap together with the VMI specification.

Reply:
@alicefr thanks for taking a look. :)

Do you recommend adding an example beyond what's in dedca6f3b2?

Reply:
I forgot to reply to [this comment](https://github.com/kubevirt/kubevirt/pull/10479#discussion_r1354498920). If Alice is thinking the same, should be to extend the current `vmi-with-generic-hook.yaml` to contain the ConfigMap so we can make the example valid with a single file.

Reply:
Ignore my previous comment. I agree with your point.

Reply:
@alicefr do you have a suggestion about how to add a configmap together with the VMI specification? I'm asking because this example is generated through code, and I don't see any examples in the `examples` directory that contain more than one resource's definition. If there was one, I could understand from it as to how to add multiple definitions in a single yaml file.

I'm still looking, but if you know it already, please point me to it. Thanks.

Reply:
Mm yeah this isn't straightforward.. I guess you will need to create a new struct that combines VMI and Configmaps and then you need to append the second object with [`MarshalOject`](https://github.com/kubevirt/kubevirt/blob/main/tools/util/marshaller.go#L32). But this can be left a TODO for a following PR

Reply:
Thanks! I added it to the `tools/vms-generator/utils/utils.go` file where the function generating this example resides.